File: esbmc.py/Testes02/validador_titulo/validador_nao_conter_caracteres_especiais_titulo.cpp
#include <cctype>

#define MAX_LEN_TITLE 21

using namespace std;

bool validate_no_special_characters(const char title[MAX_LEN_TITLE], int len) {
    const char specials[] = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`";
    for (int i = 0; i < len; i++) {
        for (int j = 0; specials[j] != '\0'; j++) {
            __ESBMC_assert(title[i] != specials[j], "O título não deve conter caracteres especiais.");
        }
    }
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];
    int len;

    __ESBMC_assume(len >= 1 && len <= 20);

    validate_no_special_characters(title, len);

    return 0;
}

File: esbmc.py/Testes02/validador_titulo/validador_nao_conter_caracteres_especiais_titulo.py
MAX_LEN_TITLE: int = 21

def validate_no_special_characters(title: str, length: int) -> bool:
    specials: str = "!@#$%^&*()_+=-[]{}|;:,.<>?/~`"

    i: int = 0
    while i < length:
        j: int = 0
        while j < len(specials):
            assert title[i] != specials[j], \
                "O título não deve conter caracteres especiais."
            j += 1
        i += 1

    return True


def main() -> None:
    title: str = input()          
    length: int = len(title)

    assert 1 <= length <= 20   

    validate_no_special_characters(title, length)


main()
